vllm/worker/hpu_worker.py

```diff
@@ -93,21 +97,82 @@ def __init__(
             torch_profiler_trace_dir = envs.VLLM_TORCH_PROFILER_DIR
             logger.info("Profiling enabled. Traces will be saved to: %s",
                         torch_profiler_trace_dir)
+
+            if os.getenv('VLLM_PROFILER_ENABLED') == 'full':
+                fn = self.full_trace_handler
+                with_stack = False
+            else:
+                fn = torch.profiler.tensorboard_trace_handler
+                with_stack = True
             self.profiler = torch.profiler.profile(
                 activities=[
                     torch.profiler.ProfilerActivity.CPU,
                     torch.profiler.ProfilerActivity.HPU,
                 ],
-                with_stack=True,
-                on_trace_ready=torch.profiler.tensorboard_trace_handler(
-                    torch_profiler_trace_dir, use_gzip=True))
+                with_stack=with_stack,
+                on_trace_ready=fn(torch_profiler_trace_dir, use_gzip=True))
         else:
             self.profiler = None
 
+    def full_trace_handler(self, dir_name, use_gzip=False):
```

Comment:
I see that this function does not use any HPUWorker class's arguments (self), so it can be extracted from HPUWorker logic and treated as a sperate. Since we already have quite a complex logic in HPUModelRunner and HPUWorker, could you move this code to vllm-hpu-extension/.../profiler.py?

Reply:
See line 133:
```
events = self.model_runner.profiler.profiling_trace_events
```